c2_store/src/mfx_c2_store.cpp

```diff
@@ -332,7 +333,19 @@ c2_status_t MfxC2ComponentStore::readXmlConfigFile()
     std::string config_filename = MFX_C2_CONFIG_XML_FILE_PATH;
     config_filename.append("/");
     config_filename.append(MFX_C2_CONFIG_XML_FILE_NAME);
-
+    ALOGI("config_filename: %s", config_filename.c_str());
+    if (config_filename.compare(config_filename.size() - 4, 4, ".xml") == 0) {
+        char codecsVariant[PROPERTY_VALUE_MAX] = {'\0'};
+        if(property_get("ro.vendor.media.target_variant_platform", codecsVariant, NULL) > 0 ) {
+            std::string variant_config_file = config_filename;
+            variant_config_file.insert(variant_config_file.size() - 4, codecsVariant);
+            struct stat fileStat;
+            if (stat(variant_config_file.c_str(), &fileStat) == 0 && S_ISREG(fileStat.st_mode)) {
+                config_filename = variant_config_file;
+                ALOGI("Changing config_filename to: %s", config_filename.c_str());
```

Comment:
MFX_DEBUG_TRACE_STREAM("Changing config_filename to: " << config_filename.c_str());